zarr/_storage/v3.py

```diff
@@ -567,6 +567,16 @@ def _normalize_store_arg_v3(store: Any, storage_options=None, mode="r") -> BaseS
         return store
     if isinstance(store, os.PathLike):
         store = os.fspath(store)
+    if FSStore._fsspec_installed():
+        import fsspec
+        if isinstance(store, fsspec.FSMap):
+            return FSStore(store.root,
+                           fs=store.fs,
+                           mode=mode,
+                           check=store.check,
+                           create=store.create,
+                           missing_exceptions=store.missing_exceptions,
+                           **(storage_options or {}))
```

Comment:
At this point, we switch from the already configured Map and internal FS instance we were given to making new instances with these storage_options, which will in general be different. This is missed in the test, which uses the local filesystem and needs no kwargs. 

(same for the other copy of this call below)

Reply:
> At this point, we switch from the already configured Map and internal FS instance we were given to making new instances with these storage_options, which will in general be different. This is missed in the test, which uses the local filesystem and needs no kwargs.

@martindurant what's your recommendation?